src/meteors/visualize/shap/plots.py

```diff
@@ -0,0 +1,691 @@
+    """a simple function that validates the explainer and explanation types.
+
+    Args:
+        explainer (HyperSHAP): an explainer object (HyperSHAP).
+        explanation (SHAPExplanation): an explanation object (SHAPExplanation).
+
+    Raises:
+        TypeError: If the explainer is not HyperSHAP type.
+        TypeError: If the explanation is not SHAPExplanation type.
+    """
+    if not isinstance(explainer, HyperSHAP):
+        raise TypeError(f"Expected HyperSHAP as explainer, but got {type(explainer)}")
+    if not isinstance(explanation, SHAPExplanation):
+        raise TypeError(f"Expected HyperSHAP as explanation, but got {type(explanation)}")
+
+
+def validate_mapping_dict(
+    mapping: dict[float | int | str, list[int] | int], explantion_values: np.ndarray, wavelengths: bool = True
+) -> dict[float | str, list[int]]:
+    """validates the mapping dictionary. The function checks if the fields of the dictionary are in the correct format, coercing it to the correct types if possible.
+    Later, it checks if all the features are used in the mapping, and returns the parsed dictionary.
+
+
+    Args:
+        mapping (dict[float | int | str, list[int] | int]): Wavelengths mapping and transformation mapping should be in a format: {band_wavelength | transformation_function_name: [feature_index, ...] | single_feature_index, ...}.
+        Transformation function name should be a string.
+        In case the wavelength is a string, there will be an attempt to convert it to a float.
+        explantion_values (np.ndarray): explanation values for the model from the shap.Explanations object.
+        wavelengths (bool, optional): If True, the function will proceed with the check for wavelengths mapping. Otherwise it will perform check for feature aggregation. Defaults to True.
+
+    Raises:
+        TypeError: Raised if mappings or explanation values are not in the correct format.
+
+    Returns:
+        mapping: dict[float, list[int]]: validated mapping dictionary.
+    """
+
+    # check if the mapping are in the correct format
+    if not isinstance(mapping, dict):
+        raise TypeError(f"Expected dict as mapping, but got {type(mapping)}")
+    if not isinstance(explantion_values, np.ndarray):
+        raise TypeError(f"Expected np.ndarray as explantion_values, but got {type(explantion_values)}")
+
+    used_indices = set()
+    ncols = explantion_values.shape[1]
+
+    parsed_mapping = {}
+
+    for key, value in mapping.items():
+        if wavelengths:
+            if isinstance(key, str):
+                try:
+                    key = float(key)
+                except ValueError:
+                    raise ValueError(f"Expected numeric as key in wavelengths, but got {key}")
+            if not isinstance(key, float) and not isinstance(key, int):
+                raise TypeError(f"Expected numeric as key in wavelengths, but got {type(key)}")
```

Comment:
Not covered by the test

Reply:
covered in 026a09dcb591956248ef1bc0ed6e165e16413e30